azuread/resource_user_test.go

```diff
@@ -0,0 +1,145 @@
+package azuread
+
+import (
+	"fmt"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azuread/azuread/helpers/ar"
+)
+
+func TestAccAzureADUser_basic(t *testing.T) {
+	resourceName := "azuread_user.test"
+	id := strconv.Itoa(rand.Intn(9999))
+	password := id + "p@$$wR2"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckADApplicationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccADUser_basic(id, password),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckADUserExists(resourceName),
+					resource.TestCheckResourceAttrSet(resourceName, "user_principal_name"),
+					resource.TestCheckResourceAttr(resourceName, "display_name", fmt.Sprintf("acctest%s", id)),
+					resource.TestCheckResourceAttr(resourceName, "mail_nickname", fmt.Sprintf("acctest%s", id)),
+					resource.TestCheckResourceAttr(resourceName, "account_enabled", "true"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"force_password_change",
+					"password", // not returned from API, sensitive
+				},
+			},
+		},
+	})
+}
+
+func TestAccAzureADUser_update(t *testing.T) {
+	resourceName := "azuread_user.test"
+	id := strconv.Itoa(rand.Intn(9999))
```

Comment:
Could we change this to: 
```suggestion
	id := acctest.RandString(7)
```